pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/SchemaConformingTransformerV2.java

```diff
@@ -402,19 +401,20 @@ private ExtraFieldsContainer processField(SchemaTreeNode parentNode, Deque<Strin
     isIndexable = isIndexable && (null == unindexableFieldSuffix || !key.endsWith(unindexableFieldSuffix));
     if (!(value instanceof Map)) {
       // leaf node
-      if (!isIndexable) {
+      if (!isIndexable && value != null) {
         extraFieldsContainer.addUnindexableEntry(key, value);
       } else {
-        if (null != currentNode && currentNode.isColumn()) {
+        if (null != currentNode && currentNode.getValue(value) != null && currentNode.isColumn()) {
```

Comment:
nit: `currentNode.getValue(value) != null` can only be true if value is null, can we early exit when value is null rather than checking in multiple places?

Reply:
done.